src/widgets/visit/visit-type.resource.ts

```diff
@@ -0,0 +1,30 @@
+import { openmrsObservableFetch } from "@openmrs/esm-api";
+import { Observable } from "rxjs";
+import { map, take } from "rxjs/operators";
+
+export function getVisitTypes(): Observable<Array<VisitType>> {
+  return openmrsObservableFetch(`/ws/rest/v1/visittype`)
+    .pipe(
+      map(results => {
+        const visitTypes: VisitType[] = results["data"][
```

Comment:
We mix quite a bit here. Personally, I always prefer `Array<T>` over `T[]`. Here, however, we use both (ln. 5, `Array<T>`, ln. 9. `T[]`).